themes/default/content/blog/api-gateway-to-eventbridge/index.md

```diff
@@ -0,0 +1,371 @@
+
+{{% /notes %}}
+
+## Create the API gateway and stage
+
+{{% choosable language typescript %}}
+
+Start by replacing the contents of `index.ts` with the following code to declare a new API Gateway [API](https://www.pulumi.com/registry/packages/aws/api-docs/apigatewayv2/api/):
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+Start by replacing the contents of `__main__.py` with the following code to declare a new API Gateway [API](https://www.pulumi.com/registry/packages/aws/api-docs/apigatewayv2/api/):
+
+{{% /choosable %}}
+
+{{< chooser language "typescript,python" / >}}
+
+{{% choosable language typescript %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+// Create an HTTP API.
+const api = new aws.apigatewayv2.Api("api", {
+    protocolType: "HTTP",
+});
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+import json
+import pulumi
+import pulumi_aws as aws
+
+# Create an HTTP API.
+api = aws.apigatewayv2.Api("example",
+    protocol_type="HTTP"
+)
+```
+
+{{% /choosable %}}
+
+Next, add a stage (feel free to name it whatever you like; I usually use the current stack name for convenience), and set it to deploy automatically whenever a change is made to the API:
+
+{{% choosable language typescript %}}
+
+```typescript
+...
+
+// Create a stage and set it to deploy automatically.
+const stage = new aws.apigatewayv2.Stage("stage", {
+    apiId: api.id,
+    name: pulumi.getStack(),
+    autoDeploy: true,
+});
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+# Create a stage and set it to deploy automatically.
+stage = aws.apigatewayv2.Stage("stage",
+    api_id=api.id,
+    name=pulumi.get_stack(),
+    auto_deploy=True
+);
+```
+
+{{% /choosable %}}
+
+The next thing to do is register a route on the gateway to give your users a publicly accessible endpoint to upload to. But in order to do _that_, you'll need to tell API Gateway what to do when an upload happens. Since the plan of record is to notify EventBridge (using API Gateway’s built-in support for it), you’ll need to declare a few EventBridge things first.
+
+## Add an event bus and event rule
+
+Every AWS account gets an event bus by default (one that's aptly named `default`), but given how easy it is to create one, we might as well do that for this application. You'll also need to define an event rule --- a resource that watches a specific event bus for events that conform to a particular pattern or shape, and then routes those events to one or more targets (e.g., Lambda functions). Add the following lines to your program for both:
+
+```typescript
```

Comment:
So if I have Python on the chooser, this part stays as Typescript. Kinda switches up my brain a bit. Intentional?

Reply:
Nope, not intentional -- I'm working on adding all the Python stuff now as well. Should be there shortly.